src/layouts/components/Editor/extensions/Cards/Cards.ts

```diff
@@ -0,0 +1,420 @@
+import { Node } from "@tiptap/core"
+import { Node as ProseMirrorNode, Schema } from "@tiptap/pm/model"
+import { Selection } from "@tiptap/pm/state"
+import { ReactNodeViewRenderer } from "@tiptap/react"
+
+import { EditorCard } from "types/editPage"
+
+import { CardsView } from "./CardsView"
+
+export interface CardOptions {
+  HTMLAttributes: {
+    class: string
+  }
+}
+
+declare module "@tiptap/core" {
+  interface Commands<ReturnType> {
+    cards: {
+      // Add a new card grid block
+      addCards: () => ReturnType
+      // Delete the current card grid block
+      deleteCards: () => ReturnType
+    }
+  }
+}
+
+interface CreateCardProps {
+  schema: Schema
+  title?: string
+  description?: string
+  footer?: string
+  link?: string
+  image?: string
+  altText?: string
+}
+
+const createCard = ({
+  schema,
+  title,
+  description,
+  footer,
+  link,
+  image,
+  altText,
+}: CreateCardProps): ProseMirrorNode => {
+  const { nodes } = schema
+  const templateCardBody = []
+  const templateCard = []
+
+  if (title && title !== "") {
+    templateCardBody.push(nodes.isomercardtitle.create({}, schema.text(title)))
+  }
+
+  if (description && description !== "") {
+    templateCardBody.push(
+      nodes.isomercarddescription.create({}, schema.text(description))
+    )
+  }
+
+  if (link && link !== "" && footer && footer !== "") {
+    templateCardBody.push(nodes.isomercardlink.create({}, schema.text(footer)))
+  }
+
+  if (image && image !== "") {
+    templateCard.push(
+      nodes.isomercardimage.create(
+        {},
+        nodes.image.create({
+          src: image,
+          alt: altText ?? "Placeholder image",
+        })
+      )
+    )
+  }
+
+  templateCard.push(nodes.isomercardbody.create({}, templateCardBody))
+
+  if (link) {
+    return nodes.isomerclickablecard.create({ href: link }, templateCard)
+  }
+
+  return nodes.isomercard.create({}, templateCard)
+}
+
+const createCardGrid = (
+  schema: Schema,
+  numberOfCards: number
+): ProseMirrorNode => {
+  const { nodes } = schema
+
+  const cards = Array(numberOfCards).fill(
+    createCard({
+      schema,
+      title: "This is a title for your card",
+      description: "This is body text for your card. Describe your card.",
+      footer: "This is a link for your card",
+      link: "https://www.isomer.gov.sg",
+      image: "https://placehold.co/600x400",
```

Comment:
hmm shall we import from local? prefer not to have an unknown dep from a third party app

Reply:
For now I keep it first? Agree with you that we should use our own placeholder image but I need to check how it would affect the rendering since technically when they save the page and it gets built on staging, the image is not on the site for the card to reference to.